crates/ruff_server/src/server.rs

```diff
@@ -64,31 +76,82 @@ impl Server {
             }
         });
 
+        let next_request_id = AtomicI32::from(1);
+
         conn.initialize_finish(id, initialize_data)?;
 
+        if dynamic_registration {
+            // Register capabilities
+            conn.sender
+                .send(lsp_server::Message::Request(lsp_server::Request {
+                    id: next_request_id.load(Ordering::Relaxed).into(),
```

Comment:
Nit: Could we use `fetch_add` here? the next request would probably need a new id anyway?